tools/test.py

```diff
@@ -721,13 +721,14 @@ def GetConfiguration(self, context):
   def GetBuildRequirements(self, path, context):
     return self.GetConfiguration(context).GetBuildRequirements()
 
-  def AddTestsToList(self, result, current_path, path, context, arch, mode):
+  def AddTestsToList(self, result, current_path, path, context,
+                     arch, mode, repeat):
     for v in VARIANT_FLAGS:
       tests = self.GetConfiguration(context).ListTests(current_path, path,
                                                        arch, mode)
       for t in tests: t.variant_flags = v
-      result += tests
-
+      for rep in range(0, repeat):
```

Comment:
0 is not needed as it is the default. Moreover xrange is preferred over range in 2.x